File: template/b1.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int solve_(int n, const std::vector<int>& arr) {
    int res = 0;
    std::vector<std::pair<int, int>> arr_with_indices(n);

    for (int i = 0; i < n; ++i) {
        arr_with_indices[i] = {arr[i], i};
    }

    std::sort(arr_with_indices.begin(), arr_with_indices.end());

    for (int i = 0; i < n; ++i) {
        int left = 0;
        int right = 0;
        int count = 0;
        for (int j = i; j < n; ++j) {
            right = std::max(right, arr_with_indices[j].second);
            if (right - left == j - i) {
                count += right - left;
                left = right;
            }
        }
        res += count;
    }

    return res;
}

int main() {
    int num_cases;
    std::cin >> num_cases;

    for (int case_num = 0; case_num < num_cases; ++case_num) {
        int n;
        std::cin >> n;

        std::vector<int> arr(n);
        for (int i = 0; i < n; ++i) {
            std::cin >> arr[i];
        }

        int res = solve_(n, arr);
        std::cout << res << std::endl;
    }

    return 0;
}

File: template/b1.py
#!/usr/bin/env python3
import sys
input = sys.stdin.readline  # to read input quickly

# available on Google, AtCoder Python3, not available on Codeforces
# import numpy as np
# import scipy

m9 = 10**9 + 7  # 998244353
yes, no = "YES", "NO"
# d4 = [(1,0),(0,1),(-1,0),(0,-1)]
# d8 = [(1,0),(1,1),(0,1),(-1,1),(-1,0),(-1,-1),(0,-1),(1,-1)]
# d6 = [(2,0),(1,1),(-1,1),(-2,0),(-1,-1),(1,-1)]  # hexagonal layout
MAXINT = sys.maxsize
e18 = 10**18 + 10

# if testing locally, print to terminal with a different color
OFFLINE_TEST = False
CHECK_OFFLINE_TEST = True
# CHECK_OFFLINE_TEST = False  # uncomment this on Codechef
if CHECK_OFFLINE_TEST:
    import getpass
    OFFLINE_TEST = getpass.getuser() == "htong"

def log(*args):
    if CHECK_OFFLINE_TEST and OFFLINE_TEST:
        print('\033[36m', *args, '\033[0m', file=sys.stderr)

def solve(*args):
    # screen input
    if OFFLINE_TEST:
        log("----- solving ------")
        log(*args)
        log("----- ------- ------")
    return solve_(*args)

def read_matrix(rows):
    return [list(map(int,input().split())) for _ in range(rows)]

def read_strings(rows):
    return [input().strip() for _ in range(rows)]

def minus_one(arr):
    return [x-1 for x in arr]

def minus_one_matrix(mrr):
    return [[x-1 for x in row] for row in mrr]

# ---------------------------- template ends here ----------------------------


def solve_(n, arr_original):
    # your solution here

    res = 0

    for i in range(n):
        for j in range(i+1, n+1):
            arr = arr_original[i:j]

            pos = {x:i for i,x in enumerate(arr)}

            brr = sorted(arr)

            crr = [pos[x] for x in brr]

            # log(crr)

            left = 0
            right = 0
            count = 0
            for i,x in enumerate(crr):
                right = max(right, x)
                if right == i:
                    # log(left, right)
                    if right > left:
                        count += right - left
                    left = i+1
            
            # log(arr, crr, count)
            res += count

    return res


for case_num in range(int(input())):

    n = int(input())
    arr = list(map(int,input().split()))
    res = solve(n, arr)  # include input here
    print(res)
